extension/src/popup/SorobanContext.tsx

```diff
@@ -31,23 +31,36 @@ export const SorobanProvider = ({
   const networkDetails = useSelector(settingsNetworkDetailsSelector);
   const source = new SorobanClient.Account(pubKey, "0");
 
-  const serverUrl =
-    networkDetails.networkPassphrase ===
-      "Test SDF Future Network ; October 2022" &&
-    networkDetails.networkUrl === FUTURENET_NETWORK_DETAILS.networkUrl
-      ? SOROBAN_RPC_URLS.FUTURENET
-      : networkDetails.networkUrl;
-
-  const server = new SorobanClient.Server(serverUrl, {
-    allowHttp: networkDetails.networkUrl.startsWith("http://"),
-  });
-
-  const newTxBuilder = () =>
-    new SorobanClient.TransactionBuilder(source, {
-      fee,
-      networkPassphrase: networkDetails.networkPassphrase,
+  let server: SorobanContextInterface['server']
+  let newTxBuilder: SorobanContextInterface['newTxBuilder']
+  if (!networkDetails.sorobanRpcUrl && networkDetails.network === NETWORKS.FUTURENET) {
+    // TODO: after enough time has passed to assume most clients have ran
+    // the migrateSorobanRpcUrlNetworkDetails migration, remove and use networkDetails.sorobanRpcUrl
+    const serverUrl = !networkDetails.sorobanRpcUrl
+      ? SOROBAN_RPC_URLS[NETWORKS.FUTURENET]
+      : networkDetails.sorobanRpcUrl
+
+    server = new SorobanClient.Server(serverUrl, {
+      allowHttp: serverUrl.startsWith("http://"),
+    });
+
+    newTxBuilder = () =>
+      new SorobanClient.TransactionBuilder(source, {
+        fee,
+        networkPassphrase: networkDetails.networkPassphrase,
+      });
+  } else if (networkDetails.sorobanRpcUrl) {
```

Comment:
pretty NIT: but should this just be an `else` rather than `else if`? We prob don't want to stumble into a scenario where `server` and `newTxBuilder` are undefined, right?